### ISSUE TITLE
Internal error related to records and type-level indices

### ISSUE THREAD

**Issue by omelkonian** [agda/agda#4148]
The minimal program below produces the following internal error:
```
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/TypeChecking/Records.hs:603
```
The type-level indices involved are particularly important:
```agda
module A where

open import Agda.Builtin.Unit using (⊤; tt)

module M1 (I : Set) where

  data T : I → Set where

  record T′ (i : I) : Set where
    constructor mk
    field
      t : T i

module M2 (I : Set) where

  open M1 I

  data R : ∀ {i} → T′ i → Set where

    r : ∀ {i} {t : T i} → R (mk t)

  test-bug : ∀ {i} {t : T′ i} → R t → ⊤
  test-bug r = tt
``` 

- Tested with `Agda version 2.6.0`

**Comment by andreasabel** [comment]
Simplified slightly to
```agda
{-# OPTIONS -v impossible:100 #-}

postulate
  A : Set

module M (I : Set) where

  postulate
    P : I → Set

  record R (i : I) : Set where
    constructor mk
    field
      f : P i

open module N = M A

data D : ∀ {i} → R i → Set where
  c : ∀ {i} {t : P i} → D (mk t)

test-bug : ∀ {i} {t : R i} → D t → Set₁
test-bug c = Set
```
Agda's last words are:
```
etaExpandRecord_: the following two constructors should be identical
  con  = Issue4148.M.R.mk
  con_ = Issue4148._.mk
```
This is a sanity check.  It may be that the check `con == con_` is too optimistic, as these constructors may not be in normal form.  Or a real problem was introduced between 2.5.4.2 and 2.6.0.  Bisect would be able to tell...
```
agda-bisect --good v2.5.4.2 --bad v2.6.0 Issue4148.agda 
```
Let's see what it says (in a couple of hours).

**Comment by andreasabel** [comment]
After fixing `agda-bisect` in #4151, I got:

8abf95c40d405c4c2ade296be23d82c80f158b1b is the first bad commit
> Author: me
> Date:   Sat Feb 23 20:09:37 2019 +0100
> 
>   [ fixed #3578 ] Less eager reduction in unifier 
> 
>   for the sake of displaying types of pattern variables to the user.

Seems the internal error is harmless, the sanity check goes through when we normalize the constructors.
